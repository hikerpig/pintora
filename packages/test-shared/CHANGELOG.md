# Change Log

## 0.2.0-alpha.3

### Patch Changes

- bc16cae: [gantt] add support for axisInterval
- 46a1a9f: [gantt] add support for axisInterval

## 0.2.0-alpha.2

### Patch Changes

- 235cc0e: add @config and @param to gantt diagram; update tmLanguage;

## 0.2.0-alpha.1

### Minor Changes

- bd214c7: Add gantt diagram

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.2-beta.0](https://github.com/hikerpig/pintora/compare/v0.1.1...v0.1.2-beta.0) (2022-02-13)

### Bug Fixes

- should support `\r` as end-of-line char and whitespace char ([f0696f4](https://github.com/hikerpig/pintora/commit/f0696f48dcb2e82b7368b8a2908dc938b6c3aea2))

# [0.1.0](https://github.com/hikerpig/pintora/compare/v0.1.0-alpha.27...v0.1.0) (2022-02-06)

**Note:** Version bump only for package @pintora/test-shared

# [0.1.0-alpha.24](https://github.com/hikerpig/pintora/compare/v0.1.0-alpha.23...v0.1.0-alpha.24) (2022-02-01)

### Features

- **pintora-diagrams:** [componentDiagram] we can have relationship from element to group ([9cc74cd](https://github.com/hikerpig/pintora/commit/9cc74cdc53013b6b63a13658ff719d414d04706f))

# [0.1.0-alpha.19](https://github.com/hikerpig/pintora/compare/v0.1.0-alpha.18...v0.1.0-alpha.19) (2022-01-22)

### Features

- **pintora-diagrams:** add mindmap support ([3062216](https://github.com/hikerpig/pintora/commit/306221660b34baad83d67435470d6aabc27b289f))

# [0.1.0-alpha.18](https://github.com/hikerpig/pintora/compare/v0.1.0-alpha.17...v0.1.0-alpha.18) (2022-01-16)

### Features

- **pintora-cli:** read cli args `--theme` ([a502c35](https://github.com/hikerpig/pintora/commit/a502c35cf1dbbad65c5847e2752e17711b8dccef))

# [0.1.0-alpha.11](https://github.com/hikerpig/pintora/compare/v0.1.0-alpha.10...v0.1.0-alpha.11) (2021-12-04)

### Features

- **pintora-diagrams:** Add basic activityDiagram [#3](https://github.com/hikerpig/pintora/issues/3) ([718a80e](https://github.com/hikerpig/pintora/commit/718a80e55d209d618fb855a463a151b880bf6fc3))

# [0.1.0-alpha.8](https://github.com/hikerpig/pintora/compare/v0.1.0-alpha.7...v0.1.0-alpha.8) (2021-08-29)

### Features

- [sequenceDiagram] add custom group color ([f93fe8a](https://github.com/hikerpig/pintora/commit/f93fe8a6fdb81af55eeb6f303f426c6150eeaf86))

# [0.1.0-alpha.4](https://github.com/hikerpig/pintora/compare/v0.1.0-alpha.3...v0.1.0-alpha.4) (2021-08-15)

### Bug Fixes

- **pintora-diagrams:** [componentDiagram] avoid interface node overlapping ([213cea0](https://github.com/hikerpig/pintora/commit/213cea090f40717390417e143366026d841f9808))

### Features

- **pintora-diagrams:** [erDiagram] add support for key coloumn like 'PK' ([64dad3d](https://github.com/hikerpig/pintora/commit/64dad3ddd8cefb912d7c83e14cf8a2dc0e64c5f8))

# [0.1.0-alpha.0](https://github.com/hikerpig/pintora/compare/v0.0.1...v0.1.0-alpha.0) (2021-08-01)

### Features

- Add componentDiagram ([c967f1c](https://github.com/hikerpig/pintora/commit/c967f1c9a969a813adac9a9967a06a024f04d73f))
