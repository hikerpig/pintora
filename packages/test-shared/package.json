{
  "name": "@pintora/test-shared",
  "version": "0.2.0-alpha.3",
  "description": "Shared testing utils for pintora",
  "keywords": [
    "pintora"
  ],
  "author": "hikerpig <hikerpigwinnie@gmail.com>",
  "homepage": "https://github.com/hikerpig/pintora#readme",
  "license": "MIT",
  "main": "lib/index.js",
  "sideEffects": false,
  "directories": {
    "lib": "lib",
    "test": "__tests__"
  },
  "files": [
    "lib"
  ],
  "publishConfig": {
    "access": "public"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/hikerpig/pintora.git"
  },
  "scripts": {
    "compile": "tsc",
    "watch": "tsc -w"
  },
  "bugs": {
    "url": "https://github.com/hikerpig/pintora/issues"
  },
  "gitHead": "92814f05b3df654b588fe35e3af275a60ea61296"
}
