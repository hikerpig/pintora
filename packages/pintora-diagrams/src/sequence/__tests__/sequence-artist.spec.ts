import pintora from '@pintora/core'
import { EXAMPLES } from '@pintora/test-shared'
import { testDraw, prepareDiagramConfig, stripDrawResultForSnapshot } from '../../__tests__/test-util'
import { sequenceDiagram } from '../index'

describe('sequence-artist', () => {
  beforeAll(() => {
    prepareDiagramConfig()
    pintora.diagramRegistry.registerDiagram('sequenceDiagram', sequenceDiagram)
  })

  it('match example snapshot', () => {
    expect(stripDrawResultForSnapshot(testDraw(EXAMPLES.sequence.code))).toMatchSnapshot()
  })
})
