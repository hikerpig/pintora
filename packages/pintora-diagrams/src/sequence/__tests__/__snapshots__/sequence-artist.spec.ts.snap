// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sequence-artist draw cross mark to 1`] = `
{
  "mark": {
    "attrs": {},
    "children": [
      {
        "children": [
          {
            "attrs": {
              "stroke": "#3b4044",
            },
            "class": "actor__line",
            "type": "line",
          },
          {
            "attrs": {
              "fill": "#fdb05e",
              "radius": 4,
              "stroke": "#3b4044",
            },
            "type": "rect",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "A",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-A",
        "type": "group",
      },
      {
        "children": [
          {
            "attrs": {
              "stroke": "#3b4044",
            },
            "class": "actor__line",
            "type": "line",
          },
          {
            "attrs": {
              "fill": "#fdb05e",
              "radius": 4,
              "stroke": "#3b4044",
            },
            "type": "rect",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "B",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-B",
        "type": "group",
      },
      {
        "attrs": {},
        "children": [],
        "class": "activations",
        "type": "group",
      },
      {
        "children": [
          {
            "attrs": {
              "lineDash": [
                3,
                3,
              ],
              "lineWidth": 2,
              "stroke": "#3b4044",
            },
            "class": "message__line",
            "itemId": "message-0-4",
            "type": "line",
          },
          {
            "attrs": {
              "lineWidth": 2,
              "stroke": "#3b4044",
            },
            "type": "path",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "stroke": "#3b4044",
              "text": "Message",
              "textAlign": "center",
              "textBaseline": "top",
            },
            "class": "message__text",
            "type": "text",
          },
        ],
        "class": "message",
        "itemId": "message-0-4",
        "type": "group",
      },
      {
        "children": [
          {
            "attrs": {
              "fill": "#fdb05e",
              "radius": 4,
              "stroke": "#3b4044",
            },
            "type": "rect",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "A",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-A",
        "type": "group",
      },
      {
        "children": [
          {
            "attrs": {
              "fill": "#fdb05e",
              "radius": 4,
              "stroke": "#3b4044",
            },
            "type": "rect",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "B",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-B",
        "type": "group",
      },
    ],
    "type": "group",
  },
}
`;

exports[`sequence-artist draw participant classifier 1`] = `
{
  "mark": {
    "attrs": {},
    "children": [
      {
        "children": [
          {
            "attrs": {
              "stroke": "#3b4044",
            },
            "class": "actor__line",
            "type": "line",
          },
          {
            "attrs": {},
            "mark": {
              "attrs": {},
              "children": [
                {
                  "attrs": {
                    "fill": "#fdb05e",
                    "lineWidth": 1.5,
                    "stroke": "#3b4044",
                  },
                  "type": "circle",
                },
                {
                  "attrs": {
                    "fill": null,
                    "lineWidth": 1.5,
                    "stroke": "#3b4044",
                  },
                  "type": "path",
                },
              ],
              "type": "group",
            },
            "type": "symbol",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "A",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-A",
        "type": "group",
      },
      {
        "children": [
          {
            "attrs": {
              "stroke": "#3b4044",
            },
            "class": "actor__line",
            "type": "line",
          },
          {
            "attrs": {},
            "mark": {
              "attrs": {},
              "children": [
                {
                  "attrs": {
                    "fill": "#fdb05e",
                    "stroke": "#3b4044",
                  },
                  "type": "path",
                },
                {
                  "attrs": {
                    "fill": "#fdb05e",
                    "stroke": "#3b4044",
                  },
                  "type": "ellipse",
                },
              ],
              "type": "group",
            },
            "type": "symbol",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "B",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-B",
        "type": "group",
      },
      {
        "children": [
          {
            "attrs": {
              "stroke": "#3b4044",
            },
            "class": "actor__line",
            "type": "line",
          },
          {
            "attrs": {},
            "mark": {
              "attrs": {},
              "children": [
                {
                  "attrs": {
                    "fill": "#fdb05e",
                    "stroke": "#3b4044",
                  },
                  "type": "path",
                },
              ],
              "type": "group",
            },
            "type": "symbol",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "C",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-C",
        "type": "group",
      },
      {
        "attrs": {},
        "children": [],
        "class": "activations",
        "type": "group",
      },
      {
        "children": [
          {
            "attrs": {},
            "mark": {
              "attrs": {},
              "children": [
                {
                  "attrs": {
                    "fill": "#fdb05e",
                    "lineWidth": 1.5,
                    "stroke": "#3b4044",
                  },
                  "type": "circle",
                },
                {
                  "attrs": {
                    "fill": null,
                    "lineWidth": 1.5,
                    "stroke": "#3b4044",
                  },
                  "type": "path",
                },
              ],
              "type": "group",
            },
            "type": "symbol",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "A",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-A",
        "type": "group",
      },
      {
        "children": [
          {
            "attrs": {},
            "mark": {
              "attrs": {},
              "children": [
                {
                  "attrs": {
                    "fill": "#fdb05e",
                    "stroke": "#3b4044",
                  },
                  "type": "path",
                },
                {
                  "attrs": {
                    "fill": "#fdb05e",
                    "stroke": "#3b4044",
                  },
                  "type": "ellipse",
                },
              ],
              "type": "group",
            },
            "type": "symbol",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "B",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-B",
        "type": "group",
      },
      {
        "children": [
          {
            "attrs": {},
            "mark": {
              "attrs": {},
              "children": [
                {
                  "attrs": {
                    "fill": "#fdb05e",
                    "stroke": "#3b4044",
                  },
                  "type": "path",
                },
              ],
              "type": "group",
            },
            "type": "symbol",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "C",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-C",
        "type": "group",
      },
    ],
    "type": "group",
  },
}
`;

exports[`sequence-artist match example snapshot 1`] = `
{
  "mark": {
    "attrs": {},
    "children": [
      {
        "attrs": {
          "lineDash": [
            2,
            2,
          ],
          "lineWidth": 2,
          "stroke": "#fdb05e",
        },
        "type": "rect",
      },
      {
        "children": [
          {
            "attrs": {
              "stroke": "#3b4044",
            },
            "class": "actor__line",
            "type": "line",
          },
          {
            "attrs": {
              "fill": "#fdb05e",
              "radius": 4,
              "stroke": "#3b4044",
            },
            "type": "rect",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "User",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-User",
        "type": "group",
      },
      {
        "children": [
          {
            "attrs": {
              "stroke": "#3b4044",
            },
            "class": "actor__line",
            "type": "line",
          },
          {
            "attrs": {
              "fill": "#fdb05e",
              "radius": 4,
              "stroke": "#3b4044",
            },
            "type": "rect",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "Pintora",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-Pintora",
        "type": "group",
      },
      {
        "attrs": {},
        "children": [
          {
            "attrs": {
              "anchor": "start",
              "fill": "#f8f8f2",
              "stroke": "#3b4044",
            },
            "class": "activation",
            "type": "rect",
          },
        ],
        "class": "activations",
        "type": "group",
      },
      {
        "children": [
          {
            "attrs": {
              "lineWidth": 2,
              "stroke": "#3b4044",
            },
            "class": "message__line",
            "itemId": "message-0-0",
            "type": "line",
          },
          {
            "attrs": {
              "fill": "#3b4044",
            },
            "type": "path",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "stroke": "#3b4044",
              "text": "render this",
              "textAlign": "center",
              "textBaseline": "top",
            },
            "class": "message__text",
            "type": "text",
          },
          {
            "attrs": {},
            "children": [
              {
                "attrs": {
                  "fill": "#3b4044",
                  "stroke": "#3b4044",
                  "symbol": "circle",
                },
                "type": "marker",
              },
              {
                "attrs": {
                  "fill": "#fdb05e",
                  "fontWeight": "bold",
                  "text": "1",
                  "textAlign": "center",
                  "textBaseline": "middle",
                },
                "class": "sequence-number",
                "type": "text",
              },
            ],
            "type": "group",
          },
        ],
        "class": "message",
        "itemId": "message-0-0",
        "type": "group",
      },
      {
        "children": [
          {
            "attrs": {
              "lineDash": [
                3,
                3,
              ],
              "lineWidth": 2,
              "stroke": "#3b4044",
            },
            "class": "message__line",
            "itemId": "message-3-1",
            "type": "path",
          },
          {
            "attrs": {
              "fill": "#3b4044",
            },
            "type": "path",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "stroke": "#3b4044",
              "text": "Has input changed?",
              "textAlign": "center",
              "textBaseline": "top",
            },
            "class": "message__text",
            "type": "text",
          },
          {
            "attrs": {},
            "children": [
              {
                "attrs": {
                  "fill": "#3b4044",
                  "stroke": "#3b4044",
                  "symbol": "circle",
                },
                "type": "marker",
              },
              {
                "attrs": {
                  "fill": "#fdb05e",
                  "fontWeight": "bold",
                  "text": "2",
                  "textAlign": "center",
                  "textBaseline": "middle",
                },
                "class": "sequence-number",
                "type": "text",
              },
            ],
            "type": "group",
          },
        ],
        "class": "message",
        "itemId": "message-3-1",
        "type": "group",
      },
      {
        "attrs": {},
        "children": [
          {
            "attrs": {
              "fill": "#fdb05e",
              "stroke": "#fdb05e",
            },
            "class": "loop__label-box",
            "type": "polygon",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "loop",
              "textBaseline": "top",
            },
            "class": "label-text",
            "type": "text",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "[Check input]",
              "textAlign": "center",
              "textBaseline": "top",
            },
            "class": "loop__title",
            "type": "text",
          },
        ],
        "class": "loop",
        "type": "group",
      },
      {
        "children": [
          {
            "attrs": {
              "lineDash": [
                3,
                3,
              ],
              "lineWidth": 2,
              "stroke": "#3b4044",
            },
            "class": "message__line",
            "itemId": "message-5-1",
            "type": "line",
          },
          {
            "attrs": {
              "fill": "#3b4044",
            },
            "type": "path",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "stroke": "#3b4044",
              "text": "your figure here",
              "textAlign": "center",
              "textBaseline": "top",
            },
            "class": "message__text",
            "type": "text",
          },
          {
            "attrs": {},
            "children": [
              {
                "attrs": {
                  "fill": "#3b4044",
                  "stroke": "#3b4044",
                  "symbol": "circle",
                },
                "type": "marker",
              },
              {
                "attrs": {
                  "fill": "#fdb05e",
                  "fontWeight": "bold",
                  "text": "3",
                  "textAlign": "center",
                  "textBaseline": "middle",
                },
                "class": "sequence-number",
                "type": "text",
              },
            ],
            "type": "group",
          },
        ],
        "class": "message",
        "itemId": "message-5-1",
        "type": "group",
      },
      {
        "children": [
          {
            "attrs": {
              "anchor": "start",
              "fill": "#F8EA75",
              "stroke": "#3b4044",
            },
            "class": "note__bg",
            "type": "rect",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "note over",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "note",
        "type": "group",
      },
      {
        "children": [
          {
            "attrs": {
              "anchor": "start",
              "fill": "#F8EA75",
              "stroke": "#3b4044",
            },
            "class": "note__bg",
            "type": "rect",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "note aside actor",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "note",
        "type": "group",
      },
      {
        "children": [
          {
            "attrs": {
              "anchor": "start",
              "fill": "#F8EA75",
              "stroke": "#3b4044",
            },
            "class": "note__bg",
            "type": "rect",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "multiline note
  is possible",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "note",
        "type": "group",
      },
      {
        "attrs": {},
        "children": [
          {
            "attrs": {
              "stroke": "#3b4044",
            },
            "type": "line",
          },
          {
            "attrs": {
              "stroke": "#3b4044",
            },
            "type": "line",
          },
          {
            "attrs": {
              "fill": "#f8f8f2",
              "lineWidth": 2,
              "stroke": "#3b4044",
            },
            "type": "rect",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 600,
              "text": "Divider",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "divider",
        "type": "group",
      },
      {
        "children": [
          {
            "attrs": {
              "fill": "#fdb05e",
              "radius": 4,
              "stroke": "#3b4044",
            },
            "type": "rect",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "User",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-User",
        "type": "group",
      },
      {
        "children": [
          {
            "attrs": {
              "fill": "#fdb05e",
              "radius": 4,
              "stroke": "#3b4044",
            },
            "type": "rect",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "Pintora",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-Pintora",
        "type": "group",
      },
      {
        "attrs": {
          "fill": "#3b4044",
          "fontFamily": "Source Code Pro, sans-serif",
          "fontSize": 16,
          "fontWeight": "bold",
          "text": "Sequence Diagram Example",
          "textAlign": "center",
        },
        "class": "sequence__title",
        "type": "text",
      },
    ],
    "type": "group",
  },
}
`;

exports[`sequence-artist will render boxes 1`] = `
{
  "mark": {
    "attrs": {},
    "children": [
      {
        "attrs": {},
        "children": [
          {
            "attrs": {
              "fill": "#d6d3fa",
              "stroke": "#3b4044",
            },
            "type": "rect",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": "bold",
              "text": "group participants",
              "textAlign": "center",
              "textBaseline": "top",
            },
            "type": "text",
          },
        ],
        "class": "sequence__participant-boxes",
        "type": "group",
      },
      {
        "children": [
          {
            "attrs": {
              "stroke": "#3b4044",
            },
            "class": "actor__line",
            "type": "line",
          },
          {
            "attrs": {
              "fill": "#fdb05e",
              "radius": 4,
              "stroke": "#3b4044",
            },
            "type": "rect",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "Alice",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-A",
        "type": "group",
      },
      {
        "children": [
          {
            "attrs": {
              "stroke": "#3b4044",
            },
            "class": "actor__line",
            "type": "line",
          },
          {
            "attrs": {
              "fill": "#fdb05e",
              "radius": 4,
              "stroke": "#3b4044",
            },
            "type": "rect",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "B",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-B",
        "type": "group",
      },
      {
        "attrs": {},
        "children": [],
        "class": "activations",
        "type": "group",
      },
      {
        "children": [
          {
            "attrs": {
              "fill": "#fdb05e",
              "radius": 4,
              "stroke": "#3b4044",
            },
            "type": "rect",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "Alice",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-A",
        "type": "group",
      },
      {
        "children": [
          {
            "attrs": {
              "fill": "#fdb05e",
              "radius": 4,
              "stroke": "#3b4044",
            },
            "type": "rect",
          },
          {
            "attrs": {
              "fill": "#3b4044",
              "fontFamily": "Source Code Pro, sans-serif",
              "fontSize": 16,
              "fontWeight": 400,
              "text": "B",
              "textAlign": "center",
              "textBaseline": "middle",
            },
            "type": "text",
          },
        ],
        "class": "actor",
        "itemId": "actor-B",
        "type": "group",
      },
    ],
    "type": "group",
  },
}
`;
